src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -1219,9 +1219,9 @@ const WorkflowView = _.flow(
     async uploadJson(key, file) {
       try {
         const rawUpdates = JSON.parse(await readFileAsText(file));
-        const updates = _.mapValues((v) => (_.isString(v) && v.match(/\${(.*)}/) ? v.replace(/\${(.*)}/, (_, match) => match) : JSON.stringify(v)))(
-          rawUpdates
-        );
+        const updates = _.mapValues((v) =>
```

Comment:
my first questions is, where are the unit tests?   Looking at this file it's HUGE and complicated... as a smaller win, this functionality should get broken out to a workflow-view-utils.ts file, give it a bit of TS love, and unit test the code this PR is specifically changing.... 

This file needs alot of TS, unit testing, and "breaking up into smaller peices" work.  But let's start small :)

Reply:
Definitely agree generally with the assessment the file seems a bit monolithic as an outsider. But as a non-DSP Eng Terra user (DSP Methods), having experienced this bug a few times just this week alone while the PR has been open, I'd appreciate being able to merge this as a quickfix if you'd be willing to approve the PR. Then maybe the workflows team can add a ticket on refactoring/adding tests for the file?

Reply:
I reached out to @mspector on Workflows team.  He and/or team will reach out on Monday and help get this to the finish line.

Reply:
Hi @rickymagner I just quickly extracted this method into a utils class and added a framework for a unit test. If you could add a couple of tests there, that would be great.

Thanks!

Reply:
Thanks for providing the template! It was really helpful given I haven't written JS tests before.

I just pushed a commit with 3 tests filled in: one "normal" parsing of a string, one parsing of a multi-line array, and one parsing of a badly formatted JSON that doesn't follow the Terra convention and gets stringified.